tests/utils/log/test_log_reader.py

```diff
@@ -98,6 +99,26 @@ def prepare_db(self, session, create_task_instance):
         )
         ti.try_number = 3
         self.ti = ti
+        dummy_ti = create_task_instance(
+            dag_id=self.DAG_ID,
+            task_id=self.DUMMY_TASK_ID,
+            start_date=self.DEFAULT_DATE,
+            run_type=DagRunType.SCHEDULED,
+            execution_date=self.DEFAULT_DATE,
+            state=TaskInstanceState.RUNNING,
+        )
+        dummy_ti.operator = 'DummyOperator'
+        self.dummy_ti = dummy_ti
+        external_task_marker_ti = create_task_instance(
+            dag_id=self.DAG_ID,
+            task_id=self.DUMMY_TASK_ID,
+            start_date=self.DEFAULT_DATE,
+            run_type=DagRunType.SCHEDULED,
+            execution_date=self.DEFAULT_DATE,
+            state=TaskInstanceState.RUNNING,
+        )
+        external_task_marker_ti.operator = 'ExternalTaskMarker'
+        self.external_task_marker_ti = external_task_marker_ti
```

Comment:
Do we need to instantiate these task instances for all tests in this class? They are only used by the four new tests and therefore should probably be handled in a separate fixture (or two fixtures for each task type).

Reply:
made separate fixtures for each task.
also had to mess with the existing fixture because `create_task_instance` fixture uses `DummyOperator` under the hood